src/Compiler/Service/TransparentCompiler.fs

```diff
@@ -258,45 +258,119 @@ module private TypeCheckingGraphProcessing =
 
             return finalFileResults, state
         }
-
-type internal CompilerCaches(sizeFactor: int) =
+type CacheSizes = {
```

Comment:
Probably should be internal

Reply:
This allows an editor to finely tweak these options. This potentially trades memory for CPU usage but on larger solutions it a tradeoff someone can make.

Reply:
I exposed these numbers so it was easier to try to adjust them individually but I do agree it's not something I'd want to expose to a user directly. 

I think the only settings come down to "memory" vs "recacluating". Specifically, my work project can be around 10-15gb once fully type checked, if I have multiple versions of open for various reasons, I'd rather try to be aggressive with collecting and keeping it's profile small.

Reply:
I guess we can expose it now for easier experimenting with some [<Experimental>] comment that it will be removed/simplified for sure. What I initially put there was pretty arbitrary but hopefully we can eventually come to some consensus here and expose only a few parameters that will be enough to tweak it.